modules/grpc/common/filterx/protobuf-field-converter.cpp

```diff
@@ -517,6 +518,156 @@ class BytesFieldConverter : public ProtobufFieldConverter
   }
 };
 
+FilterXObject *
+MapFieldConverter::get(Message *message, ProtoReflectors reflectors)
+{
+  const gchar *key_name = "key";
+  const gchar *value_name = "value";
+
+  FilterXObject *dict = filterx_dict_new();
+
+  int len = reflectors.reflection->FieldSize(*message, reflectors.fieldDescriptor);
+  for (int i = 0; i < len; i++)
+    {
+      Message *elem_message = reflectors.reflection->MutableRepeatedMessage(message, reflectors.fieldDescriptor, i);
+      ProtoReflectors key_reflectors(*elem_message, key_name);
+      ProtoReflectors value_reflectors(*elem_message, value_name);
+
+      ProtobufFieldConverter *key_converter = get_protobuf_field_converter(key_reflectors.fieldType);
+      ProtobufFieldConverter *value_converter = get_protobuf_field_converter(value_reflectors.fieldType);
+
+      FilterXObject *key_object = key_converter->get(elem_message, key_name);
+      if (!key_object)
+        return NULL;
+
+      FilterXObject *value_object = value_converter->get(elem_message, value_name);
+      if (!value_object)
+        {
+          filterx_object_unref(key_object);
+          return NULL;
+        }
+
+      if (!filterx_object_set_subscript(dict, key_object, &value_object))
+        {
+          filterx_object_unref(key_object);
+          filterx_object_unref(value_object);
+          return NULL;
+        }
+
+      filterx_object_unref(key_object);
+      filterx_object_unref(value_object);
+    }
+
+  return dict;
+}
+
+gboolean
+MapFieldConverter::add_elem(FilterXObject *key, FilterXObject *value, gpointer user_data)
+{
+  google::protobuf::Message *message = static_cast<google::protobuf::Message *>(((gpointer *) user_data)[0]);
+  ProtoReflectors *reflectors = static_cast<ProtoReflectors *>(((gpointer *) user_data)[1]);
+
+  const gchar *key_name = "key";
+  const gchar *value_name = "value";
+
+  try
+    {
+      Message *elem_message = reflectors->reflection->AddMessage(message, reflectors->fieldDescriptor);
+      ProtoReflectors key_reflectors(*elem_message, key_name);
+      ProtoReflectors value_reflectors(*elem_message, value_name);
+
+      ProtobufFieldConverter *key_converter = get_protobuf_field_converter(key_reflectors.fieldType);
+      ProtobufFieldConverter *value_converter = get_protobuf_field_converter(value_reflectors.fieldType);
+
+      FilterXObject *assoc_key_object = NULL;
+      if (!key_converter->set(elem_message, key_name, key, &assoc_key_object))
+        return FALSE;
+
+      filterx_object_unref(assoc_key_object);
+
+      FilterXObject *assoc_val_object = NULL;
+      if (!value_converter->set(elem_message, value_name, value, &assoc_val_object))
+        return FALSE;
+
+      filterx_object_unref(assoc_val_object);
+    }
+  catch (const std::exception &e)
```

Comment:
Optional: more specific exception type + reduced try-catch scope here and similar places?